### ISSUE TITLE
Ruby 1.9.* rake Errors

### ISSUE THREAD

**Issue by loganfranken** [ucla-oarc-web/WebBlocks#200]
While executing the `rake` command with Ruby 1.9.2 or 1.9.3 (most recent distributions from RubyInstaller for Windows), I ran into a few issues (works just fine with 1.8.7):
#### Error 1

**Error:** `cannot load such file -- lib/Rake`
**Solution:** Change `require 'lib/Rake'` to `require './lib/Rake'`
#### Error 2

**Error:** `uninitialized constant VERSION`
**Solution:** Change `VERSION` to `RUBY_VERSION` in `Ruby193\lib\ruby\gems\1.9.1\gems\extensions-0.6.0\lib\extensions\_base.rb`. (Is there a newer version of this gem that wouldn't have this problem?)
#### Error 3

**Error:** `Ruby/Extensions: internal error: was supposed to implement Kernel#require_relative, but it didn't!`
**Solution:** Didn't attempt to resolve this one as I figure there must be some rabbit hole of incompatibilities with this gem and Ruby 1.9.*.

**Comment by ebollens** [comment]
Assigning this to myself to resolve. I'm sure there are more issues with WebBlocks builder at Ruby 1.9.x that will only show up when running the unit tests (like the use of `String.respond_to? :each`).

**Comment by ebollens** [comment]
I'm currently working on trying to get RVM set up on OS X. Unfortunately, because XCode has gotten rid of GCC and Ruby 1.8 requires GCC to compile, it's a tremendous mess.

**Comment by ebollens** [comment]
The problems described at the start of this issue have been solved.

However, these are not the only issues with Ruby 1.9.3 versus 1.8.7. Instead, once resolving these, others are becoming apparent. Will post issues encountered and resolutions as they occur.

```
[Rake] Load task definitions
rake aborted!
cannot load such file -- build.rb
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:17:in `require'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:17:in `block (4 levels) in define_tasks'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:65:in `block in info'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:87:in `scope'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:64:in `info'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:16:in `block (3 levels) in define_tasks'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:14:in `each'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:14:in `block (2 levels) in define_tasks'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:13:in `chdir'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:13:in `block in define_tasks'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:12:in `chdir'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake/Manager.rb:12:in `define_tasks'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake.rb:53:in `block in run'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:40:in `block in task'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:87:in `scope'
/Applications/MAMP/htdocs/WebBlocks/lib/Logger.rb:39:in `task'
/Applications/MAMP/htdocs/WebBlocks/lib/Rake.rb:52:in `run'
/Applications/MAMP/htdocs/WebBlocks/Rakefile:31:in `<top (required)>'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/rake_module.rb:25:in `load'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/rake_module.rb:25:in `load_rakefile'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:583:in `raw_load_rakefile'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:89:in `block in load_rakefile'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:160:in `standard_exception_handling'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:88:in `load_rakefile'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:72:in `block in run'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:160:in `standard_exception_handling'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/lib/rake/application.rb:70:in `run'
/usr/local/rvm/gems/ruby-1.9.3-p194/gems/rake-10.0.3/bin/rake:33:in `<top (required)>'
/usr/local/rvm/gems/ruby-1.9.3-p194/bin/rake:23:in `load'
/usr/local/rvm/gems/ruby-1.9.3-p194/bin/rake:23:in `<main>'
```

**Comment by ebollens** [comment]
On this branch, `rake` now successfully completes with Ruby 1.8.7 and Ruby 1.9.3. The unit tests all still pass in Ruby 1.8.7 as well, but there are 5 failures and 1 error occurring for the unit tests in Ruby 1.9.3.

The unit tests failing currently for Ruby 1.9.3:

```

  1) Error:
test_resolve_sass_dependencies(TestUnitModule):
NoMethodError: undefined method `each' for "m2":String
    /Applications/MAMP/htdocs/WebBlocks/lib/Build/Module.rb:178:in `block in resolve_sass_dependencies'
    /Applications/MAMP/htdocs/WebBlocks/lib/Build/Module.rb:173:in `each'
    /Applications/MAMP/htdocs/WebBlocks/lib/Build/Module.rb:173:in `resolve_sass_dependencies'
    /Applications/MAMP/htdocs/WebBlocks/test/unit/Module.rb:91:in `test_resolve_sass_dependencies'

  2) Failure:
test_build_dispatcher(TestUnitRakeDSL) [/Applications/MAMP/htdocs/WebBlocks/test/unit/RakeDSL.rb:38]:
Failed assertion, no message given.

  3) Failure:
test_build_manager(TestUnitRakeDSL) [/Applications/MAMP/htdocs/WebBlocks/test/unit/RakeDSL.rb:44]:
Failed assertion, no message given.

  4) Failure:
test_config(TestUnitRakeDSL) [/Applications/MAMP/htdocs/WebBlocks/test/unit/RakeDSL.rb:20]:
Failed assertion, no message given.

  5) Failure:
test_logger(TestUnitRakeDSL) [/Applications/MAMP/htdocs/WebBlocks/test/unit/RakeDSL.rb:26]:
Failed assertion, no message given.

  6) Failure:
test_rake_manager(TestUnitRakeDSL) [/Applications/MAMP/htdocs/WebBlocks/test/unit/RakeDSL.rb:32]:
Failed assertion, no message given.
```

**Comment by ebollens** [comment]
The error (1) is caused by the fact that Ruby 1.9 does not provide an `.each` method for `String`.

This is solved by forcing to array as:

``` ruby
config[:src][:modules].respond_to? :each ? config[:src][:modules] : [config[:src][:modules]]
```

While the compile I ran `rake` with didn't have this sort of a modules configuration, this is definitely something that would have bit us - hence the benefit of unit tests.

**Comment by ebollens** [comment]
The other bug appears to be caused by the fact that `assert_method_exists` is not provided in 1.9.3. It is merely a bug with the unit tests, therefore, than with the actual compiler. However, need to fix this for a good weather report on 1.9.3.

**Comment by ebollens** [comment]
All unit and integration tests now pass for both Ruby 1.8.7 and Ruby 1.9.3, so this is ready for pull request merge.
